src/huggingface_hub/hf_api.py

```diff
@@ -1423,22 +1422,25 @@ def upload_file(
             raise ValueError(f"Invalid repo type, must be one of {REPO_TYPES}")
 
         if token is None:
-            token = self._validate_or_retrieve_token()
-        elif not self._is_valid_token(token):
-            if self._is_valid_token(path_or_fileobj):
-                warnings.warn(
-                    "`upload_file` now takes `token` as an optional positional argument. "
-                    "Be sure to adapt your code!",
-                    FutureWarning,
-                )
-                token, path_or_fileobj, path_in_repo, repo_id = (
-                    path_or_fileobj,
-                    path_in_repo,
-                    repo_id,
-                    token,
-                )
-            else:
-                raise ValueError("Invalid token passed!")
+            token = self._validate_or_retrieve_token(function_name="upload_file")
```

Comment:
Should this be updated since the logic below is in `validate_or_retrieve_token`?